test/benchmark/main.cc

```diff
@@ -1,16 +1,36 @@
 // NOLINT(namespace-envoy)
 // This is an Envoy driver for benchmarks.
+#include "envoy/common/platform.h"
 
 #include "benchmark/benchmark.h"
 
 #ifdef ENVOY_HANDLE_SIGNALS
 #include "common/signal/signal_action.h"
 #endif
+#include "common/common/assert.h"
 
 #include "absl/debugging/symbolize.h"
 
+#if defined(WIN32)
+static void NoopInvalidParameterHandler(const wchar_t* expression, const wchar_t* function,
+                                        const wchar_t* file, unsigned int line,
+                                        uintptr_t pReserved) {
+  return;
+}
+#endif
+
 // Boilerplate main(), which discovers benchmarks and runs them.
 int main(int argc, char** argv) {
+#if defined(WIN32)
+  _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
+
+  _set_invalid_parameter_handler(NoopInvalidParameterHandler);
+
+  WSADATA wsa_data;
+  const WORD version_requested = MAKEWORD(2, 2);
+  RELEASE_ASSERT(WSAStartup(version_requested, &wsa_data) == 0, "");
+#endif
```

Comment:
This is repeated somewhere else. Can you have a common win32 utility function that you call from the mains?

Reply:
Seems like this block is also repeated a lot: https://github.com/envoyproxy/envoy/blob/11eecae656d868be92540e0276e60fc2e80efe88/test/fuzz/main.cc#L57-L63

We could pull all of this out into a "test main init" somewhere in the test utils if that makes sense?

Reply:
Sure any cleanup appreciated.